ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js

```diff
@@ -11,245 +11,254 @@
-                    break;
-              case "GEOMETRYCOLLECTION":
-                    shapes = locationGeometry.geometries;
-                    locationFilter = this.buildIntersectOrCQL(shapes);
-                    break;
-              default:
-                    console.log("unknown location type");
-                    return;
-            }
+                        if (i !== shapes.length - 1) {
+                              locationFilter += " OR ";
+                        }
+                    }.bind(this));
+                }
+                else {
+                    locationFilter = "(INTERSECTS(anyGeo, " + locationWkt + "))";
+                }
+                break;
+          case "MULTIPOINT":
+                shapes = locationGeometry.points;
+                locationFilter = this.buildIntersectOrCQL(shapes);
+                break;
+          case "MULTIPOLYGON":
+                shapes = locationGeometry.polygons;
+                locationFilter = this.buildIntersectOrCQL(shapes);
+                break;
+          case "MULTILINESTRING":
+                shapes = locationGeometry.lineStrings;
+                locationFilter = this.buildIntersectOrCQL(shapes);
+                break;
+          case "GEOMETRYCOLLECTION":
+                shapes = locationGeometry.geometries;
+                locationFilter = this.buildIntersectOrCQL(shapes);
+                break;
+          default:
+                console.log("unknown location type");
+                return;
+        }
 
-            return locationFilter;
-        },
-        buildIntersectOrCQL: function(shapes){
-            var locationFilter = "";
-            $.each(shapes, function (i, shape) {
-                locationFilter += this.buildIntersectCQL(shape);
+        return locationFilter;
+    },
+    buildIntersectOrCQL: function(shapes){
+        var locationFilter = "";
+        $.each(shapes, function (i, shape) {
+            locationFilter += this.buildIntersectCQL(shape);
 
-                if (i !== shapes.length - 1) {
-                      locationFilter += " OR ";
-                }
-            }.bind(this));
+            if (i !== shapes.length - 1) {
+                  locationFilter += " OR ";
+            }
+        }.bind(this));
 
-            return locationFilter;
-        },
-        arrayFromCQLGeometry: function(cql) {
-            // remove opening 'POLYGON(' or 'MULTIPOLYGON(' as well as closing ')'
-            var result = cql.replace(/^\w+\(/, "").replace(/\)$/, "");
-            // change parentheses to array brackets
-            result = result.replace(/\(/g,'[').replace(/\)/g,']');
-            // change each space-separated coordinate pair to a two-element array
-            result = result.replace(/([^,\[\]]+)\s+([^,\[\]]+)/g, '[$1,$2]');
-            // build nested arrays from the string
-            return JSON.parse(result);
+        return locationFilter;
+    },
+    arrayFromPartialWkt: function(partialWkt) {
```

Comment:
❓  If we don't expect anything outside this file to use this function, we should consider not exporting it.

Reply:
I removed all the local helper methods from the exports